tests/common.py

```diff
@@ -0,0 +1,199 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Contains several commonly used objects (models, dataloaders) that are shared across the test suite."""
+
+from typing import Sequence
+
+import pytest
+import torch
+from PIL import Image
+from torch.utils.data import Dataset
+from torchvision.datasets import VisionDataset
+
+from composer.core.precision import Precision
+from composer.models import ComposerClassifier
+
+# syntactic sugar below
+# decorators for common parameterizations and marks
+
+
+def device(*args, precision=False):
```

Comment:
I am hesitant here because that would hide the settings being tested to the test builder/debugger -- they would need to see the decorators to understand what is going on, which defeats the purpose of the syntactic sugar, which is to reduce syntax but keep needed information.

I agree that it is doing too much. I'll file an issue to clean this up, perhaps split into two decorators:
```
@device('gpu', 'cpu')
```
```
@device_precision('gpu,fp32', 'gpu,amp')
```

The decorator names should also convey the fixture names (e.g. device, and device_precision). Would that work?

Reply:
Agreed, that makes sense. Thinking about it more, I'm wondering if we need to hide the underlying parameterization logic, since that makes an assumption about what the field name is called. Would something like this instead work?

# common.py
```python3
device_cpu = pytest.param(DeviceCPU())
device_gpu = pytest.param(DeviceGPU(), mark=pytest.mark.gpu)
```
Then, the test case would do something like:
```python3
@pytest.mark.paramtereize(device, [common.device_cpu, common.device_gpu])
def test_something(device: Device):
    ...
```

or for device/precision:
```python3
@pytest.mark.paramtereize("device,precision", [(common.device_gpu, Precision.FP32), (common.device_gpu, Precision.AMP])
def test_something_else(device: Device, precision: Precision):
    ...
```